integration/image/pull_test.go

```diff
@@ -155,3 +158,48 @@ func TestImagePullStoredfDigestForOtherRepo(t *testing.T) {
 	assert.Assert(t, err != nil, "Expected error, got none: %v", err)
 	assert.Assert(t, errdefs.IsNotFound(err), err)
 }
+
+// TestImagePullNonExisting pulls non-existing images from the central registry, with different
+// combinations of implicit tag and library prefix.
+func TestImagePullNonExisting(t *testing.T) {
+	ctx := setupTest(t)
+
+	for _, tc := range []struct {
+		alias string
+		tag   string
+	}{
```

Comment:
This struct could have been “ref” and “all”

Reply:
Actually `ref` alone is enough here. Done.